src/components/lib/cos_component.c

```diff
@@ -21,16 +27,80 @@ cos_init(void *arg)
 	main();
 }
 
+/* Intended to be implement by libraries */
 CWEAKSYMB void
-__alloc_libc_initilize(void)
+pre_syscall_default_setup()
+{
+}
+
+/* Intended to be overriden by components */
+CWEAKSYMB void
+pre_syscall_setup()
+{
+	pre_syscall_default_setup();
+}
+
+CWEAKSYMB void
+syscall_emulation_setup()
+{
+}
+
+CWEAKSYMB long
+cos_syscall_handler(int syscall_num, long a, long b, long c, long d, long e, long f, long g)
+{
+	printc("Default syscall handler callled (syscall: %d), faulting!", syscall_num);
+	assert(0);
+	return 0;
+}
+
+__attribute__((regparm(1))) long
+__cos_syscall(int syscall_num, long a, long b, long c, long d, long e, long f, long g)
+{
+	return cos_syscall_handler(syscall_num, a, b, c, d, e, f, g);
+}
+
+CWEAKSYMB void
+libc_initialization_handler()
 {
-	return;
+}
+
+/* TODO: Make this a weak symbol (currently doing so makes this fail) */
+void __init_libc(char **envp, char *pn);
+
+void
+libc_init()
+{
+	/* The construction of this is:
+	 * evn1, env2, ..., NULL, auxv_n1, auxv_1, auxv_n2, auxv_2 ..., NULL
+	 * TODO: Figure out a way to set AT_HWCAP / AT_SYSINFO
+	 */
+	static char *envp[] = {
```

Comment:
Is this the minimal set?  This is all essentially wasted memory, and isn't doing anything useful (but it is the easiest way to work with libc).  Thus minimizing its impact seems reasonable.